solidity/contracts/converter/types/standard-pool/StandardPoolConverter.sol

```diff
@@ -247,32 +220,56 @@ contract StandardPoolConverter is
     }
 
     /**
-     * @dev withdraws tokens held by the converter and sends them to an account
+     * @dev transfers reserves to a new converter during an upgrade
+     * can only be called by the converter upgraded which should be set at its owner
+     *
+     * @param _newConverter account to receive the new amount
+     */
+    function transferReservesOnUpgrade(address _newConverter)
+        external
+        override
+        protected
+        ownerOnly
+        only(CONVERTER_UPGRADER)
+    {
+        uint256 reserveCount = __reserveTokens.length;
+        for (uint256 i = 0; i < reserveCount; ++i) {
+            IERC20 reserveToken = __reserveTokens[i];
+
+            uint256 amount;
+            if (reserveToken == NATIVE_TOKEN_ADDRESS) {
+                amount = address(this).balance;
+            } else {
+                amount = reserveToken.balanceOf(address(this));
+            }
+
+            safeTransfer(reserveToken, _newConverter, amount);
+
+            syncReserveBalance(reserveToken);
+        }
+    }
+
+    /**
+     * @dev withdraws inactive reserves tokens held by the converter and sends them to an account
      * can only be called by the owner
-     * note that reserve tokens can only be withdrawn by the owner while the converter is inactive
-     * unless the owner is the converter upgrader contract
      *
-     * @param _token   ERC20 token contract address
-     * @param _to      account to receive the new amount
-     * @param _amount  amount to withdraw
+     * @param _token ERC20 token contract address
+     * @param _to account to receive the new amount
+     * @param _amount amount to withdraw
      */
-    function withdrawTokens(
+    function withdrawInactiveTokens(
         IERC20 _token,
         address _to,
         uint256 _amount
-    ) public override(IConverter, TokenHolder) protected ownerOnly {
-        address converterUpgrader = addressOf(CONVERTER_UPGRADER);
+    ) public override(IConverter) protected ownerOnly {
         uint256 reserveId = __reserveIds[_token];
 
-        // if the token is not a reserve token, allow withdrawal
-        // otherwise verify that the converter is inactive or that the owner is the upgrader contract
-        require(reserveId == 0 || !isActive() || owner == converterUpgrader, "ERR_ACCESS_DENIED");
-        super.withdrawTokens(_token, _to, _amount);
+        // verify that the token is a reserve token and that the converter is inactive
+        require(reserveId != 0 && !isActive(), "ERR_ACCESS_DENIED");
 
-        // if the token is a reserve token, sync the reserve balance
-        if (reserveId != 0) {
-            syncReserveBalance(_token);
-        }
+        safeTransfer(_token, _to, _amount);
+
+        syncReserveBalance(_token);
```

Comment:
same comment as the one in the ConverterBase

Reply:
👍